### PR TITLE
Fix amount<->percent conversions on v2 payout modal 

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

- Fix amount<->percent conversions by using `feePercentage` in converter functions 
- Adds `currencyContext` to `projectPayoutMods` to make it actually version-agnostic
- Moves `PayoutModsList` back to v1 since it's got a lot of v1 stuff and not currently used in v2 

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/v2/V2Create/tabs/FundingTab/FundingTabContent.tsx [hunk 1]
```diff
@@ -245,8 +250,8 @@ export default function FundingTabContent({
           <ProjectPayoutMods
             mods={splits.map(toMod)}
             target={target ?? '0'}
-            currency={toV1Currency(targetCurrency)}
-            feePerbicent={ETHPaymentTerminalFee}
+            currencyName={targetCurrency === ETH ? 'ETH' : 'USD'}
+            feePercentage={formatFee(ETHPaymentTerminalFee)}
```
Comment (aeolianeth) [inline src/components/v2/V2Create/tabs/FundingTab/FundingTabContent.tsx:254]:
```suggestion
            feePercentage={ETHPaymentTerminalFee ? formatFee(ETHPaymentTerminalFee) : undefined}
```

File: src/utils/v2/math.ts [hunk 2]
```diff
@@ -99,7 +99,8 @@ export const redemptionRateFrom = (percentage: string) => {
  * @param feePerBillion - fee as parts-per-billion.
  * @returns {string} fee expressed as a percentage.
  */
-export const formatFee = (feePerBillion: BigNumber) => {
+export const formatFee = (feePerBillion: BigNumber | undefined) => {
+  if (!feePerBillion) return
```
Comment (aeolianeth) [inline src/utils/v2/math.ts:103]: I kinda think this pattern throughout the codebase is a bit of a code small. I think e.g. we should conditionally render `ProjectPayoutMods` if `ETHPaymentTerminalFee` is defined, or at least do what I suggested in the above comment.

Let me know what you think

**Review by aeolianeth — CHANGES_REQUESTED** [review]
@johnnyd-eth thanks heaps for fixing this, great catch and this solution is 👌 

couple comments

File: src/utils/v2/math.ts (hunk 2 above)
Comment (johnnyd-eth) [reply src/utils/v2/math.ts:103]: Bit on the nose I agree. Will change

File: src/components/shared/forms/PayModsForm.tsx [hunk 3]
```diff
@@ -97,14 +102,14 @@ export default function PayModsForm({
         <FormItems.ProjectPayoutMods
           mods={mods}
           target={fromWad(target)}
-          currency={currency}
+          currencyName={currency === ETH ? 'ETH' : 'USD'}
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/shared/forms/PayModsForm.tsx:105]: @aeolianeth This is a shared component, so I don't think so. But for the v1 specific components, yep

File: src/components/v1/PayoutModsList.tsx [hunk 4]
```diff
@@ -228,8 +237,10 @@ export default function PayoutModsList({
               lockedMods={lockedMods}
               onModsChanged={setEditingMods}
               target={fromWad(fundingCycle.target)}
-              currency={fundingCycle.currency.toNumber() as V1CurrencyOption}
-              feePerbicent={feePerbicent}
+              currencyName={
+                fundingCycle.currency.toNumber() === ETH ? 'ETH' : 'USD'
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v1/PayoutModsList.tsx:241]: Yep

File: src/utils/v2/math.ts (hunk 2 above)
Comment (aeolianeth) [reply src/utils/v2/math.ts]: hahahahaha

Comment (aeolianeth) [reply src/utils/v2/math.ts]: bit pongy

**Review by aeolianeth — APPROVED** [review]

**Pull request merged**